URAP-charter/web_scraping
# Distributed Crawling -- Non-API Run

The purpose of this branch is to allow us to run a full crawl of the 7k charter school list while navigating around the various roadblock urls.

This new branch uses Docker to split a list of urls into 100-line chunks, then runs each chunk in parallel using Python's multiprocessing package.

To test:
Follow the directions in the README!
1. Run Mongo
2. From scrapy/schools, run "docker build . -t schoolcrawler"
3. Run "docker ps" to get the id of the created container
4. Run "docker run --network='host' containerid"
5. Follow the logs (they terminate randomly, but the spider is still running): docker logs containerid --follow
6. Check Mongo: docker exec -it mongodb bash, mongo -u username -p password, show dbs
7. See how much data you can get!
